api/document/serializers/doc_cursor.py

```diff
@@ -49,31 +49,65 @@ def to_internal_value(self,
         ]
 
 
-class DocCursorSerializer(serializers.ModelSerializer):
+def docnode_field_attrs(field: str, attrs: List[str]) -> Dict[str, Any]:
+    """
+    Translate Django model field attributes from the DocNode model
+    into attributes that Django REST Framework serializers
+    understand.
+
+    Note that we're doing this manually instead of using
+    serializers.ModelSerializer because:
+
+        * DocCursor isn't actually a Django model.  It's *like*
+          a model in a lot of ways, but its just different
+          enough that annoying workarounds would be needed
+          to make it work with ModelSerializer.
+
+        * There are enough deviations between the Django model's
+          validation and the REST API's validation that we'd
+          still have to write a bunch of code to "undo"
+          ModelSerializer's defaults.
+    """
+
+    attr_map = {
+        'blank': 'allow_blank',
+    }
+    metadata = DocNode._meta.get_field(field)
+    result = {
+        attr_map.get(attr, attr): getattr(metadata, attr)
+        for attr in attrs
+    }
+    if result.get('allow_blank'):
+        result['required'] = False
+    return result
```

Comment:
Blerg, I guess this helper is confusing/bloaty enough that we can just get rid of it and do things more manually/explicitly.  I just wanted to reduce DRY by not repeating stuff from the model too much...

Reply:
Any chance we can leverage utilities from DRF? I agree that this is probably more confusing that being redundant.

Reply:
Unfortunately, not as far as I can tell. It's ok though, I'm just gonna get rid of this funky function and make things more explicit while still preserving DRY as much as possible.